src/librustc_codegen_llvm/intrinsic.rs

```diff
@@ -686,6 +682,36 @@ impl IntrinsicCallMethods<'tcx> for Builder<'a, 'll, 'tcx> {
         let expect = self.get_intrinsic(&"llvm.expect.i1");
         self.call(expect, &[cond, self.const_bool(expected)], None)
     }
+
+    fn va_start(&mut self, list: &'ll Value) -> &'ll Value {
+        let target = &self.cx.tcx.sess.target.target;
+        let arch = &target.arch;
+        let va_list = match (&**arch, target.options.is_like_windows) {
+            (_, true) => list,
+            ("aarch4", _) if target.target_os == "ios" => list,
+            ("aarch4", _) | ("x86_64", _) | ("powerpc", _) =>
+                self.load(list,
+                          self.tcx().data_layout.pointer_align.abi),
+            _ => list
+        };
+        let intrinsic = self.cx().get_intrinsic("llvm.va_start");
+        self.call(intrinsic, &[va_list], None)
+    }
+
+    fn va_end(&mut self, list: &'ll Value) -> &'ll Value {
+        let target = &self.cx.tcx.sess.target.target;
+        let arch = &target.arch;
+        let va_list = match (&**arch, target.options.is_like_windows) {
+            (_, true) => list,
+            ("aarch4", _) if target.target_os == "ios" => list,
+            ("aarch4", _) | ("x86_64", _) | ("powerpc", _) =>
+                self.load(list,
+                          self.tcx().data_layout.pointer_align.abi),
+            _ => list,
+        };
```

Comment:
If you make the first case `_ if target.options.is_like_windows`, you can avoid using a tuple for the remaining cases.

Reply:
Done